core/src/main/scala/akka/kafka/javadsl/Producer.scala

```diff
@@ -42,6 +58,22 @@ object Producer {
     scaladsl.Producer.commitableSink(settings)
       .mapMaterializedValue(_.toJava)
       .asJava
+  /**
+   * Sink that is aware of the [[ConsumerMessage#CommittableOffset committable offset]]
+   * from a [[Consumer]]. It will commit the consumer offset when the message has
+   * been published successfully to the topic.
+   *
+   * Note that there is always a risk that something fails after publishing but before
+   * committing, so it is "at-least once delivery" semantics.
+   */
+  def commitableSink[K, V](
+    producerProvider: () => KafkaProducer[K, V],
+    closeTimeout: FiniteDuration,
+    parallelism: Int
```

Comment:
I looked some more at this. Having specific parameters like this will be difficult to maintain in a compatible way. It's better to pass in the `ProducerSettings`, which can easier be evolved.